src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1446,6 +1446,7 @@
     <!-- Non-null type is required for this node kind -->
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
     <Field Name="Property" Type="PropertySymbol" Null="disallow"/>
+    <Field Name="PropertyIndex" Type="int"/>
```

Comment:
> <Field Name="PropertyIndex" Type="int"/> [](start = 3, length = 41)

I am not sure if we need to store the index, it should always match the index of this item in Declarations array in BoundAnonymousObjectCreationExpression

Reply:
For `new {a, b = i }`, Declarations array only has a single entry for property `b`, but `PropertyIndex` is 1. How can we figure out that the first element of `Arguments` has no matching element in `Declarations` array without using this property index?

---
In reply to: [196900867](https://github.com/dotnet/roslyn/pull/28005#discussion_r196900867) [](ancestors = 196900867)

Reply:
> For new {a, b = i }, Declarations array only has a single entry for property b, but PropertyIndex is 1. How can we figure out that the first element of Arguments has no matching element in Declarations array without using this property index?

I see.



---
In reply to: [196904027](https://github.com/dotnet/roslyn/pull/28005#discussion_r196904027) [](ancestors = 196904027,196900867)